Zend/zend_ast.c

```diff
@@ -436,7 +436,7 @@ static int zend_ast_add_array_element(zval *result, zval *offset, zval *expr)
 			zend_hash_index_update(Z_ARRVAL_P(result), zend_dval_to_lval(Z_DVAL_P(offset)), expr);
 			break;
 		case IS_RESOURCE:
-			zend_error(E_NOTICE, "Resource ID#%d used as offset, casting to integer (%d)", Z_RES_HANDLE_P(offset), Z_RES_HANDLE_P(offset));
+			zend_error(E_WARNING, "Resource ID#%d used as offset, casting to integer (%d)", Z_RES_HANDLE_P(offset), Z_RES_HANDLE_P(offset));
 			zend_hash_index_update(Z_ARRVAL_P(result), Z_RES_HANDLE_P(offset), expr);
 			break;
 		default:
```

Comment:
I think I'll have to leave this out, because when I tried it, I ended up having test issues. This is what I did BTW:
```php
if (Z_TYPE_P(offset) == IS_UNDEF) {
	if (!zend_hash_next_index_insert(Z_ARRVAL_P(result), expr)) {
		zend_throw_error(NULL, "Cannot add element to the array as the next element is already occupied");
		return FAILURE;
	}
	return SUCCESS;
}

return array_set_zval_key(Z_ARRVAL_P(result), offset, expr);

```

Reply:
I think I'll have to leave this out, because when I tried it, I ended up having test issues. This is what I did BTW:
```php
if (Z_TYPE_P(offset) == IS_UNDEF) {
	if (!zend_hash_next_index_insert(Z_ARRVAL_P(result), expr)) {
		zend_throw_error(NULL, "Cannot add element to the array as the next element is already occupied");
		return FAILURE;
	}
	return SUCCESS;
}

return array_set_zval_key(Z_ARRVAL_P(result), offset, expr);

```